tokio/src/runtime/handle.rs

```diff
@@ -252,6 +252,14 @@ impl Handle {
     /// [`tokio::time`]: crate::time
     #[track_caller]
     pub fn block_on<F: Future>(&self, future: F) -> F::Output {
+        #[cfg(all(
```

Comment:
Similarly, many of these cfgs could be reduced to a single `#[cfg(tokio_taskdump)]` since `lib.rs` contains a `compile_error!` if you have `#[cfg(tokio_taskdump)]` without the other options also set.

Reply:
I'm concerned that reducing these to just `#[cfg(tokio_taskdump)]` wouldn't prevent errors in these locations under odd configurations, and thus would drown out the lone explicit `compile_error` in `lib.rs`. I think Carl's suggestion is probably the way to go.